src/context.rs

```diff
@@ -436,8 +442,48 @@ impl Context {
     pub fn emit_event(&self, event: EventType) {
         self.events.emit(Event {
             id: self.id,
-            typ: event,
+            typ: event.clone(),
         });
+        // We use `task::spawn()` which could mix up the order of events in the db and
+        // could make e.g. backups fail since it could still run in the
+        // background while trying to move/close the database.
```

Comment:
What can actually happen when using tokio::spawn() @Hocuri?
The original solution does no longer work since we removed async_std from the project and using task::spawn was one of your suggestions on the initial pr.

Reply:
- it makes stuff slower
- Back then, it made one of the import/export tests flaky because (IIRC)
  - the test called stop_io()
  - one of the spawned tasks was still running
  - this task tried accessing the db
  - the db was busy while it should be exported

Did you read https://github.com/deltachat/deltachat-core-rust/pull/3296#discussion_r1041483869 yet?

Reply:
No I didn't, thank you for pointing that out!

Reply:
The problem with a channel is that sending to it has to be async and I don't really see where a good place for going async is...